onnxruntime/core/framework/session_state.cc

```diff
@@ -254,7 +254,7 @@ Status SessionState::GeneratePatternGroupCache(const std::vector<std::reference_
             len *= dim.dim_value();
           }
         }
-        if (!IAllocator::CalcMemSizeForArrayWithAlignment<64>(len, ml_data_type->Size(), &size)) {
+        if (!IAllocator::CalcMemSizeForArrayWithAlignment<64>(static_cast<size_t>(len), ml_data_type->Size(), &size)) {
```

Comment:
i wonder if we should use SafeInt here, i think that was introduced recently and is intended for allocation related calculations

Reply:
Please.